### DISCUSSION TITLE
`fromEvent` in v7

### DISCUSSION THREAD

**Discussion by pratikgaikwadhive** [ReactiveX/rxjs#6450] [Q&A] [answered]
We use `rethinkdb` in our project. In v6 of rxjs I could write
```
const event$ = rxjs.merge<Changes<IGuest>, RethinkDBError>(
      rxjs.fromEvent<Changes<IGuest>>(feed, "data"),
      // convert error events to errors.
      rxjs.fromEvent<RethinkDBError>(feed, "error").pipe(rxOperator.tap(err => { throw err; }))
    );
```
where `Changes` and `RethinkDBError` are from Rethinkdb-ts. 

Now same thing is broken in v7.
![image](https://user-images.githubusercontent.com/36861239/120846127-b2e69700-c53f-11eb-96b2-14f1f8bfbd1e.png)

`fromEvent` complains that it can't resolve either `Changes<IGuest>>` or `RethinkDBerror`. 

Can someone please help me in resolving this? What is the correct use for `fromEvent` in server side coding?

**Comment by backbone87** [comment, marked as answer]
you should not set generic type parameters explicitly:
```
/** @deprecated Do not specify explicit type parameters. Signatures with type parameters that cannot be inferred will be removed in v8. */
```
you should remove them:
```ts
const event$ = rxjs.merge(
  rxjs.fromEvent(feed, "data"),
  // convert error events to errors.
  rxjs.fromEvent(feed, "error").pipe(rxOperator.tap(err => { throw err; }))
);
```
ofc this only works, if `feed` event listeners are properly typed with overloads for different events. if not, then you probably need to resort to a type cast:
```ts
const event$ = rxjs.merge(
  rxjs.fromEvent(feed, "data"),
  // convert error events to errors.
  rxjs.fromEvent(feed, "error").pipe(rxOperator.tap(err => { throw err; }))
) as Observable<Changes<IGuests>>;
```

**Reply by pratikgaikwadhive** [reply]
@backbone87,

Thank you for replying. `changes` comes from [rethinkdb-ts](https://github.com/rethinkdb/rethinkdb-ts) which is completely written in TS. `IGuest` is from my internal project which is also written in TS. When I hover my mouse over `feed` it shows correct type. But let me try your second option

**Reply by backbone87** [reply]
from a quick look at rethinkdb-ts it seems like they just plainly extend `EventEmitter` without overloading its methods with proper event specific typing. you can test this very easily yourself:
```ts
feed.on('data', (...args) => {
  /* args should be properly typed here over your domain */
});
```

so I guess you must resort to typecasting the observable.

**Reply by cartant** [reply]
The type parameter was removed because it had no type-safe purpose. It was nothing more than a type assertion in disguise and type assertions are better represented using the proper syntax - `as Observable<Changes<IGuests>>` - as Oliver pointed out above.

**Reply by pratikgaikwadhive** [reply]
Thanks all, I will be using type casting for now.
